### PR TITLE
feat: email notification integration

### PR DESCRIPTION
**Pull request by mlorek**
feat: email notification integration - #1693

## Test Plan

* [ x] acceptance tests

## References
https://docs.snowflake.com/en/user-guide/email-stored-procedures

*

### CODE REVIEW

File: pkg/resources/email_notification_integration_acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	resource.Test(t, resource.TestCase{
+		Providers:    providers(),
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: emailNotificationIntegrationConfig(emailIntegrationName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_email_notification_integration.test", "name", emailIntegrationName),
+				),
```
Comment (sfc-gh-swinkler) [inline pkg/resources/email_notification_integration_acceptance_test.go:23]: you need to add an import test. Something like:

			// IMPORT
	```
		{
				ResourceName:     "snowflake_email_notification_integration.test",
				ImportState:       true,
				ImportStateVerify: true,
			},
```

**Review by sfc-gh-swinkler — CHANGES_REQUESTED** [review]
looks good, just a few minor changes requested

File: pkg/resources/email_notification_integration.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	// The first part of the schema is shared between all integration vendors
+	"name": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"enabled": {
+		Type:     schema.TypeBool,
+		Required: true,
+		// Default:  true,
```
[the remark this answers is not in the document]
Comment (mlorek) [reply pkg/resources/email_notification_integration.go]: 'Default cannot be set with Required'

**Comment by sfc-gh-swinkler** [comment]
Thanks @mlorek, this looks good!.  I have been working on a new design for the Snowflake SDK to replace the "snowflake" package. You can learn more about it here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1767. Would appreciate if you had any feedback / would be able to contribute using this new design pattern in the future.

**Review by sfc-gh-swinkler — APPROVED** [review]
Thanks @mlorek !

**Pull request merged**